src/sidebar/components/menu-item.js

```diff
@@ -91,7 +91,7 @@ export default function MenuItem({
   let focusTimer = null;
 
   let renderedIcon = null;
-  if (icon !== 'blank') {
+  if (icon && icon !== 'blank') {
```

Comment:
Sure a typeof `undefined` is not equal to `string`, but the check (`!==`) is actually asking if the type is not equal in the first place. It just seems like a redundant check and we don't need to check if icon is `undefined` first. Or if we did, then this should be valid:

`if (icon && icon != 'blank')`

but then the linter complains! I feel like TS does not help in cases like this. Granted they are fewer than not, but I think it's important to see the forest for the trees with this typing stuff because it can become a burden at times.

Reply:
The check isn't redundant here - if icon is `undefined|string` then inside the `if` statement it could still be `undefined` or any string other than `"blank"`. This value was then being passed to a prop that requires a string. That's what TS was correctly complaining about. I agree that strict equality is not actually necessary here - it's just a convention to use it everywhere.

> Granted they are fewer than not, but I think it's important to see the forest for the trees with this typing stuff because it can become a burden at times.

I agree it is good to be pragmatic about these things and not let too much typing noise creep into the code.